sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/JsonSerializable.java

```diff
@@ -526,9 +527,23 @@ private String toPrettyJson(Object object){
      * @return the POJO.
      */
     public <T> T toObject(Class<T> c) {
+        // TODO: We have to remove this if we do not want to support CosmosItemProperties anymore, and change all the
+        //  tests accordingly
+        if(CosmosItemProperties.class.isAssignableFrom(c)){
+            return (T) new CosmosItemProperties(this.toJson());
+        }
+        
         if (JsonSerializable.class.isAssignableFrom(c) || String.class.isAssignableFrom(c)
                 || Number.class.isAssignableFrom(c) || Boolean.class.isAssignableFrom(c)) {
-            throw new IllegalArgumentException("c can only be a POJO class or JSONObject");
+            return c.cast(this.get(Constants.Properties.VALUE));
```

Comment:
Do we have tests for Number values?
If not please add tests

Reply:
Integer is a Number. I tested Integer with both Number and Integer and it works